src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -81,6 +122,11 @@ public Retry getRetry(Method method) {
     return method.getDescriptor().getMethodAnnotation(AnnotationsProto.retry);
   }
 
+  /** Return the extra retry codes for the given method. */
+  public String getResourceType(Field field) {
+    return (String) field.getOptionFields().get(AnnotationsProto.resourceType.getDescriptor());
```

Comment:
Converting null to empty here will make checks easier upstream.

Reply:
Hmm, that would hide null values behind empty strings, and potentially hide NPEs.

Also, I would prefer to have the calling code do the `Strings.isNullOrEmpty()` check. That way, if I forget to convert empty to null in one of the ProtoParser methods, the calling method is still fine.

Reply:
Good point